libs/libc/termios/lib_cfspeed.c

```diff
@@ -0,0 +1,196 @@
+{
+  speed_t idx = 0;
+
+  for (; idx < ARRAYSIZE(g_baud_table); idx++)
+    {
+      if (speed == g_baud_table[idx])
+        {
+          break;
+        }
+    }
+
+  /* we don't find the speed value, it could be mask */
+
+  if (idx == ARRAYSIZE(g_baud_table))
+    {
+      return (speed & ~CBAUD) ? -1 : speed;
+    }
+
+  /* If idx > B38400, we should will idx minus 15, and or CBAUDEX */
+
+  if (idx > B38400)
+    {
+      idx -= B38400;
+      idx |= CBAUDEX;
+    }
+
+  return idx;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: cfsetspeed
+ *
+ * Description:
+ *   The cfsetspeed() function is a non-POSIX function that sets the baud
+ *   stored in the structure pointed to by termiosp to speed.
+ *
+ *   There is no effect on the baud set in the hardware until a subsequent
+ *   successful call to tcsetattr() on the same termios structure.
+ *
+ *   NOTE 1: NuttX does not control input/output baud independently.  Both
+ *   must be the same.  The POSIX standard interfaces, cfisetispeed() and
+ *   cfisetospeed() are defined to be cfsetspeed() in termios.h.
+ *
+ *   NOTE 3: A consequence of NOTE 1 is that you should never attempt to
+ *   set the input and output baud to different values.
+ *
+ *   Also, the following POSIX requirement cannot be supported: "If the input
+ *   baud rate stored in the termios structure pointed to by termios_p is 0,
+ *   the input baud rate given to the hardware will be the same as the output
+ *   baud rate stored in the termios structure."
+ *
+ *   NOTE 2. In Nuttx, the speed_t is defined to be unsigned int and the baud
+ *   encodings of termios.h are baud value mask. And their corresponding
+ *   values are in array g_baud_table. However, if you do so, your code will
+ *   *NOT* be portable to other environments where speed_t is smaller and
+ *   where the termios.h baud values are encoded! To avoid portability
+ *   issues, use the baud definitions in termios.h!
+ *
+ *   Linux, for example, would require this (also non-portable) sequence:
+ *
+ *     cfsetispeed(termiosp, BOTHER);
+ *     termiosp->c_ispeed = baud;
+ *
+ *     cfsetospeed(termiosp, BOTHER);
+ *     termiosp->c_ospeed = baud;
+ *
+ * Input Parameters:
+ *   termiosp - The termiosp argument is a pointer to a termios structure.
+ *   speed - The new input speed. It could be baud rate or could be mask.
+ *
+ * Returned Value:
+ *   Baud is returned. If speed don't match g_baud_table and mask in
+ *   termios.h, -1 is returned and set errno EINVAL.
+ *
+ ****************************************************************************/
+
+int cfsetspeed(FAR struct termios *termiosp, speed_t speed)
+{
+  int mask = baud_mask(speed);
+
+  DEBUGASSERT(termiosp);
+  if (mask == -1)
+    {
+      set_errno(-EINVAL);
```

Comment:
remove -

Reply:
Yes, I have finished the modification. Thanks!